File: typescript/behavioral/strategy/operators/context.ts
import {Strategy} from "./strategy";

export class Context {
  private strategy: Strategy

  constructor(strategy: Strategy) {
    this.strategy = strategy
  }

  public setStrategy(strategy: Strategy): void {
    this.strategy = strategy
  }

  public operation(a: number, b: number): number {
    return this.strategy.execute(a, b);
  }
}

File: dart/behavioral/strategy/operators/bin/context.dart
import 'strategy.dart';

class Context {
  Strategy _strategy;

  Context(Strategy s) {
    _strategy = s;
  }

  void setStrategy(Strategy s) {
    _strategy = s;
  }

  double operation(double a, double b) {
    return _strategy.execute(a, b);
  }
}
